core/src/test/scala/unit/kafka/server/AbstractFetcherThreadTest.scala

```diff
@@ -929,6 +929,29 @@ class AbstractFetcherThreadTest {
     fetcher.verifyLastFetchedEpoch(partition, Some(5))
   }
 
+  @Test
+  def testMaybeUpdateTopicIds(): Unit = {
+    val partition = new TopicPartition("topic1", 0)
+    val fetcher = new MockFetcherThread
+
+    // Start with no topic IDs
+    fetcher.setReplicaState(partition, MockFetcherThread.PartitionState(leaderEpoch = 0))
+    fetcher.addPartitions(Map(partition -> initialFetchState(None, 0L, leaderEpoch = 0)))
+
+
+    def verifyFetchState(fetchState: Option[PartitionFetchState], expectedTopicId: Option[Uuid]): Unit = {
+      assertTrue(fetchState.isDefined)
+      assertEquals(expectedTopicId, fetchState.get.topicId)
+    }
+
+    verifyFetchState(fetcher.fetchState(partition), None)
+
+    // Add topic ID
+    fetcher.maybeUpdateTopicIds(Set(partition), topicName => topicIds.get(topicName))
```

Comment:
Should we also test few negative cases?

Reply:
I can add a case where the partition doesn't exist.

I thought of a potential case where the ID wasn't zero, but we don't handle that at this level (we will always write the ID regardless of the current value.) Should we change this? I'm trying to think if we will ever overwrite a value. Since this is synchronous I don't think so.

Reply:
Any other cases I'm missing? I suppose there is also zero topic ID in the map. But that is similar to the potential case I mentioned. It is handled earlier and shouldn't occur if we are on this code path. (If we do end up here, we will write the zero ID)

Reply:
To clarify, this path is only taken if 
```
requestTopicId match {
                  case Some(topicId) if logTopicId.isEmpty =>
```
meaning both the request topic ID and log topic ID are non-empty/non-zero UUID.

So just to clarify, should I still check these edge cases (either log ID is already defined/ID in the map is zero)

Reply:
We could add tests for the ReplicaManager which verifies that the topic id is updated and propagated to the fetcher manager. Is it what you are thinking about?

Regarding this particular test, what would you add? If the topic id is zero, we would just set it. As you said, the logic which prevents this from happening is before.